crates/hdk/src/hash_path/path.rs

```diff
@@ -269,47 +252,115 @@ impl From<String> for Path {
 }
 
 impl Path {
-    pub fn path_entry(&self) -> ExternResult<PathEntry> {
-        Ok(PathEntry::new(hash_entry(self)?))
+    /// Attach a [`LinkType`] to this path
+    /// so its type is known for [`create_link`] and [`get_links`].
+    pub fn into_typed(self, link_type: impl Into<LinkType>) -> TypedPath {
+        TypedPath::new(link_type, self)
     }
 
+    /// Try attaching a [`LinkType`] to this path
+    /// so its type is known for [`create_link`] and [`get_links`].
+    pub fn try_into_typed<TY, E>(self, link_type: TY) -> Result<TypedPath, WasmError>
```

Comment:
I need to still keep the into typed though because it's used in path in places where an error can't happen. I could change `try_into_typed` -> `typed` though